store/sqlstore/channel_member_history_store.go

```diff
@@ -86,17 +86,66 @@ func (s SqlChannelMemberHistoryStore) GetUsersInChannelDuring(startTime int64, e
 	})
 }
 
-func (s SqlChannelMemberHistoryStore) PurgeHistoryBefore(time int64, channelId string) store.StoreChannel {
+func (s SqlChannelMemberHistoryStore) PermanentDeleteBatch(endTime int64, limit int64) store.StoreChannel {
```

Comment:
This method can't be easily tested, because it trashes data that other tests depend on, so I broke out the bit that actually deletes data into `PermanentDeleteBatchForChannel(...)`, since that can be tested because it only deletes data for a specific channel id.

Reply:
Why do the other tests depend on the data?

Reply:
Excellent question. It's not really that other tests depend on this data, but that other tests make an unknown amount of data, depending on the order that the suite runs in. This code deletes a specified number of records from the table in no particular order, but doesn't know how many records already exist, so it's difficult to assert that this code deleted the correct number of records without adding a store method that just returns the entire contents of the table, which I don't want to do, because it makes a mess of the interface.

So for example, if the other tests created a bunch of join/leave events, and then this test ran, we would expect to delete, say, 100 records. But if this test ran before those other tests, we would expect to delete just the number one record that this test creates before calling the delete method.

That's why I broke this code out into two methods - the other one takes a channel id, and channel ids are unique, so I can assert that the correct number of records exist for the channel id before and after running this code.

Reply:
Would it be possible to test it by deleting all of the data using this method, creating some new data, then deleting a subset of that data using this method?

Reply:
Alternatively, it's totally okay to use SQL directly from the tests. You don't have to add anything to the interface just to inspect the table beforehand.

Reply:
On second thought, you're absolutely right. I needlessly complicated this code. Fixed in https://github.com/mattermost/mattermost-server/pull/7937/commits/53454f262bc78419af260c957e0b4f4877a2c3cb